lib/maplibre-compose-material3/src/commonMain/kotlin/dev/sargunv/maplibrecompose/material3/controls/ScaleBar.kt

```diff
@@ -1,253 +1,300 @@
-import androidx.compose.animation.EnterTransition
-import androidx.compose.animation.ExitTransition
-import androidx.compose.animation.core.MutableTransitionState
-import androidx.compose.animation.fadeIn
-import androidx.compose.animation.fadeOut
 import androidx.compose.foundation.Canvas
-import androidx.compose.foundation.layout.Arrangement
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.size
 import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.contentColorFor
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableDoubleStateOf
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.geometry.Offset
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.StrokeCap
-import androidx.compose.ui.unit.Dp
-import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.rememberTextMeasurer
 import androidx.compose.ui.unit.dp
-import dev.sargunv.maplibrecompose.compose.CameraState
+import androidx.compose.ui.unit.toSize
+import dev.sargunv.maplibrecompose.material3.defaultScaleBarMeasure
+import dev.sargunv.maplibrecompose.material3.drawPathsWithHalo
+import dev.sargunv.maplibrecompose.material3.drawTextWithHalo
 import dev.sargunv.maplibrecompose.material3.generated.Res
 import dev.sargunv.maplibrecompose.material3.generated.feet_symbol
 import dev.sargunv.maplibrecompose.material3.generated.kilometers_symbol
 import dev.sargunv.maplibrecompose.material3.generated.meters_symbol
 import dev.sargunv.maplibrecompose.material3.generated.miles_symbol
-import kotlin.math.roundToInt
-import kotlin.time.Duration
-import kotlin.time.Duration.Companion.seconds
-import kotlinx.coroutines.delay
 import org.jetbrains.compose.resources.stringResource
 
-public data class ScaleBarColors(
-  val textColor: Color,
-  val lineColor: Color,
-  val shadowColor: Color,
-)
-
-public object ScaleBarDefaults {
-  public val width: Dp = 65.dp
-  public val height: Dp = 50.dp
-
-  @Composable
-  public fun colors(): ScaleBarColors =
-    ScaleBarColors(
-      textColor = MaterialTheme.colorScheme.onSurface,
-      lineColor = MaterialTheme.colorScheme.onSurface,
-      shadowColor = MaterialTheme.colorScheme.surface,
-    )
+/** Which measure to show on the scale bar. */
+public enum class ScaleBarMeasure {
+  /** meters / kilometers */
+  Metric,
+  /** feet / miles */
+  Imperial,
+  /** both feet / miles and meters / kilometers */
+  ImperialAndMetric;
```

Comment:
Haven't reviewed the whole thing yet but just a naming nit: the "imperial" measuring system means specifically the UK system while the US system is "US customary". Both evolved from an older English system, and today differ in some units (though feet and miles are the same for both, ignoring the "survey foot" and "survey mile" and nonsense like that).

I'd name this `Meters`, `Miles`, and `MetersAndMiles` or something like that, and update comments referencing "imperial" to reference "feet / miles".

Reply:
Meters and Miles sounds off, because one mile is roughly 1500x of a meter.

I've looked over the wikipedia articles for US customary units, imperial units etc.

The gist of it is that regarding (feet + mile) length, the Imperial and US customary measurement system are identical through the [international yard and pound](https://en.wikipedia.org/wiki/International_yard_and_pound) agreement of 1959. So, the full name would be "US customary and British Imperial measurement units".

Suggestions:
- leave as is
- use `Meters`, `FeetAndMiles`, `Both`
- use `SI`, `US`, `UK` (like in CLDR). However, the CLDR `MeasurementSystem` refers to the measurement system in general, including weights, masses, areas, volues, ... , and is thus much more coarse.

Reply:
Addendum: According to the CLDR and confirmed by viewing some StreetView imagery, in the United Kingdom, they don't use feet+miles, but yards+miles ... 🤔

Reply:
Perhaps the enum has three options: `Metric`, `FeetAndMiles`, `YardsAndMiles` (or SI, US, UK), and the scale bar takes up to two of them as parameters: `primaryMeasurementSystem` and `secondaryMeasurementSystem`.

* Default in most of the world: primary is SI, secondary is null
* Default everywhere else,
  * primary is US or UK if not transitioning, SI if transitioning
  * secondary is SI if not transitioning, US or UK if transitioning